cmd/operator-sdk/alpha/bundle/validate.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bundle
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/operator-framework/operator-registry/pkg/lib/bundle"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+// newBundleBuildCmd returns a command that will build operator bundle image.
+func newBundleValidateCmd() *cobra.Command {
+	c := bundleCmd{}
+	cmd := &cobra.Command{
+		Use:   "validate",
+		Short: "Validate an operator bundle image",
+		Long: `The 'operator-sdk bundle validate' command will validate both content and
+format of an operator bundle image containing operator metadata and manifests.
+This command will exit with a non-zero exit code if any validation tests fail.`,
+		Example: `The following invocation will validate a test-operator bundle image hosted
+in a Docker image registry:
+
+$ operator-sdk bundle validate quay.io/example/operator:v0.1.0 --image-builder docker`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 1 {
+				return fmt.Errorf("command %s requires exactly one argument", cmd.CommandPath())
+			}
+			c.imageTag = args[0]
+
+			dir, err := ioutil.TempDir("", "bundle-")
+			if err != nil {
+				return err
+			}
+			defer func() {
+				if err = os.RemoveAll(dir); err != nil {
+					log.Error(err.Error())
+				}
+			}()
+			logger := log.WithFields(log.Fields{"container-tool": c.imageBuilder})
+			log.SetLevel(log.DebugLevel)
+			val := bundle.NewImageValidator(c.imageBuilder, logger)
+			if err = val.PullBundleImage(c.imageTag, dir); err != nil {
+				log.Fatal(err)
+			}
+
+			logger.Info("Validating bundle image format and contents")
+
+			if err = val.ValidateBundleFormat(dir); err != nil {
+				log.Fatal(err)
+			}
+			manifestsDir := filepath.Join(dir, bundle.ManifestsDir)
+			if err = val.ValidateBundleContent(manifestsDir); err != nil {
+				log.Fatal(err)
+			}
+
+			logger.Info("All validation tests have completed successfully")
```

Comment:
Suggestion: extract for something such as: `val.ValidateBundle()`

Reply:
That isn't easily done since we need to defer cleanup of `dir`. I think complexity is low enough to not refactor into separate functions.